### PR TITLE
[lldb] Add the SwiftEnableBareSlashRegexOnPlayground setting

### PR DESCRIPTION
**Pull request by augusto2112**
rdar://92690194

### CODE REVIEW

**Comment by augusto2112** [comment]
@swift-ci test

**Comment by augusto2112** [comment]
@adrian-prantl should this be enabled or disabled by default? I left it disabled in the current patch.

**Comment by augusto2112** [comment]
@swift-ci test

**Comment by augusto2112** [comment]
@swift-ci test

File: lldb/source/Target/TargetProperties.td [hunk 1]
```diff
@@ -19,6 +19,9 @@ let Definition = "target_experimental" in {
   def SwiftDiscoverImplicitSearchPaths: Property<"swift-discover-implicit-search-paths", "Boolean">,
     DefaultTrue,
     Desc<"Discover implicit search paths from all implicitly imported Swift modules and make them available to the expression context. A Swift module built with -serialize-debugging-options can contain additional search paths which are discovered as the module is imported. This optiondoes an eager import of all modules first to make sure all implicit search paths are availableto the expression evaluator. If the build system registers all Swift modules with the linker (Darwin: via -add_ast_path, Other platforms: -module-wrap), turning this on is not necessary.">;
+  def SwiftEnableBareSlashRegexOnPlayground: Property<"swift-enable-bare-slash-regex-playground", "Boolean">,
```
Comment (adrian-prantl) [inline lldb/source/Target/TargetProperties.td]: Nit: there's nothing playground-specific about this patch. Maybe we should drop the playground from all names.

**Comment by augusto2112** [comment]
@swift-ci test

**Comment by augusto2112** [comment]
@swift-ci test

**Comment by augusto2112** [comment]
@swift-ci test

**Review by adrian-prantl — APPROVED** [review]
Thanks!

**Comment by adrian-prantl** [comment]
@swift-ci test macos

**Pull request closed without merging**